hazelcast/src/main/java/com/hazelcast/internal/blackbox/impl/BlackboxImpl.java

```diff
@@ -0,0 +1,321 @@
+                sensors.put(parameter, sensor);
+            }
+
+            checkOverwrite(parameter, sensor);
+
+            if (logger.isFinestEnabled()) {
+                logger.finest("Registered sensor " + parameter);
+            }
+
+            sensor.source = source;
+            sensor.input = input;
+            if (sensor.view != null) {
+                sensor.view.input = input;
+            }
+
+            modCount.incrementAndGet();
+            return sensor;
+        }
+    }
+
+    /**
+     * We are going to check if a source or input already exist. If it exists, we are going the log a warning but we
+     * are not going to fail. The blackbox should never fail unless the arguments don't make any sense of course.
+     *
+     * @param parameter
+     * @param sensor
+     */
+    private void checkOverwrite(String parameter, SensorImpl sensor) {
+        // if an input already exists, we are just going to overwrite it.
+        if (sensor.input != null) {
+            logger.warning(format("Duplicate registration, a SensorInput for parameter '%s' already exists", parameter));
+        }
+
+        // if a source already exists, we are just going to overwrite it.
+        if (sensor.source != null) {
+            logger.warning(format("Duplicate registration, a source for parameter '%s' already exists", parameter));
+        }
+    }
+
+    @Override
+    public SensorView getView(String parameter) {
+        checkNotNull(parameter, "parameter can't be null");
+
+        synchronized (getLock(parameter)) {
+            SensorImpl sensor = sensors.get(parameter);
+
+            if (sensor == null) {
+                sensor = new SensorImpl(parameter, this);
+                sensors.put(parameter, sensor);
+            }
+
+            if (sensor.view != null) {
+                return sensor.view;
+            }
+
+            SensorViewImpl view = new SensorViewImpl(this, parameter, logger);
+            sensor.view = view;
+            view.input = sensor.input;
+            return view;
+        }
+    }
+
+    public void destroy(String parameter) {
+        checkNotNull(parameter, "parameter can't be null");
+
+        synchronized (getLock(parameter)) {
+            SensorImpl sensor = sensors.remove(parameter);
+            if (sensor == null) {
+                return;
+            }
+
+            if (sensor.view != null) {
+                sensor.view.input = null;
+            }
+            sensor.input = null;
+            sensor.view = null;
+
+            if (logger.isFinestEnabled()) {
+                logger.finest("Destroying sensor " + parameter);
+            }
+
+            modCount.incrementAndGet();
+        }
+    }
+
+    @Override
+    public void scheduleAtFixedRate(final Runnable publisher, long delay, TimeUnit timeUnit) {
+        Runnable publishTask = new Runnable() {
+            @Override
+            public void run() {
+                publisher.run();
+            }
+        };
```

Comment:
Good spotting. I had an interface SensorPublisher with a publish method and I needed to wrap that in a runnable to get executed. I got rid of the interface and replaced it with a Runnable. So that is the story of the runnable in a runnable. I'll fix it